packages/apps/job-launcher/server/src/modules/payment/payment.entity.ts

```diff
@@ -1,20 +1,21 @@
-import { Column, Entity, JoinColumn, ManyToOne } from 'typeorm';
+import { Column, Entity, Index, JoinColumn, ManyToOne } from 'typeorm';
 import { NS } from '../../common/constants';
 import { BaseEntity } from '../../database/base.entity';
-import {
-  Currency,
-  PaymentSource,
-  PaymentType,
-} from '../../common/enums/payment';
+import { PaymentSource, PaymentType } from '../../common/enums/payment';
 import { UserEntity } from '../user/user.entity';
 
-@Entity({ schema: NS, name: 'payment' })
+@Entity({ schema: NS, name: 'payments' })
+@Index(['chainId', 'transaction'], {
+  unique: true,
+  where: '(chain_Id IS NOT NULL)',
+})
+@Index(['transaction'], { unique: true, where: '(chain_Id IS NULL)' })
```

Comment:
Take care, because some payment records don't have any transaction id, for example the we want to withdraw some amount from user balance.